src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -77,10 +81,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
-
-    Message message = new Message(user, text);
+    float sentimentScore = getSentimentScore(text);
+    
+    Message message = new Message(user, text, sentimentScore);
     datastore.storeMessage(message);
 
     response.sendRedirect("/user-page.html?user=" + user);
   }
+
+  /**
+   * Analyzes a message's text for its positive/negative sentiment.
+   *
+   * @return the sentiment score of the message
+   */
+  private float getSentimentScore(String text) throws IOException {
```

Comment:
Just out of curiosity, what throws this? Is this something we can handle differently? (If there's an error while fetching the sentiment, maybe default it to 0 or something?)

Reply:
I thought it could be thrown during document parsing, but all I can find in my confusing trek through the docs is that `languageService.close()` can throw a generic `Exception` and that `languageService.analyzeSentiment(doc)` can throw `com.google.api.gax.rpc.ApiException` if the remote API call fails. 

I think that actually came straight out of the starter code -- I suspect it's not necessary to keep it, but I'm going to add more robust error checking/handling in several parts tomorrow morning so I'll check it out further then.

Reply:
Honestly I wouldn't worry too too much about it. Here's what I would do:

Try taking out the `throws IOException` part. If it doesn't complain, you're good.

If it does complain, **maybe** wrap whatever it's complaining about in a `try / catch` block.

If that ends up being too messy, just stick with the `throws IOException` part.

I'd rather you have time to spend exploring cool things rather than spend it all on one particular exception case!